File: core/r/entry/render.R
#!/usr/bin/env Rscript
# Entry: figkit render. Rebuild a figure from an edited plot_config.yaml.
suppressPackageStartupMessages(library(optparse))

opts <- list(
  make_option("--config", type = "character"),
  make_option("--data",   type = "character", default = NULL),
  make_option("--sheet",  type = "character", default = NULL),
  make_option("--stamp",  type = "character", default = NULL),
  make_option("--engine", type = "character", default = NULL),  # read by the figkit dispatcher for routing; ignored here
  make_option("--out",    type = "character", default = "pubready_output")
)
opt <- parse_args(OptionParser(option_list = opts))
if (is.null(opt$config)) stop("missing required --config")

source(file.path(Sys.getenv("PUBREADY_CORE", "/opt/pubready/core"), "r", "bootstrap.R"))

spec <- spec_from_config(opt$config)
raw_input <- opt$data %||% spec$data$file
if (is.null(raw_input)) stop("no input data: pass --data or set data.file in the config")

run_recipe(spec, raw_input = raw_input, out_root = opt$out, sheet = opt$sheet,
           stamp = opt$stamp)

File: core/python/entry/render.py
#!/usr/bin/env python3
"""Entry: figkit render (Python engine)."""
import argparse
import os
import sys

sys.path.insert(0, os.path.join(os.environ.get("PUBPLOT_CORE", "/opt/pubplot/core"), "python"))
from pubplot.run import run_recipe          # noqa: E402
from pubplot.spec import spec_from_config    # noqa: E402


def main():
    p = argparse.ArgumentParser()
    p.add_argument("--config", required=True)
    p.add_argument("--data"); p.add_argument("--sheet"); p.add_argument("--stamp")
    p.add_argument("--engine")
    p.add_argument("--out", default="pubplot_output")
    a, _ = p.parse_known_args()
    spec = spec_from_config(a.config)
    raw = a.data or spec.get("data", {}).get("file")
    if not raw:
        raise SystemExit("no input data: pass --data or set data.file in the config")
    run_recipe(spec, raw, a.out, sheet=a.sheet, stamp=a.stamp)


if __name__ == "__main__":
    main()
